### ISSUE TITLE
[Windows]  GetExitCodeProcess() -> STILL_ACTIVE

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#1663]
Occurs very rarely.
```
======================================================================
FAIL: psutil.tests.test_process.TestPopen.test_ctx_manager
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\cygwin64\home\user\_psutil\psutil\tests\test_process.py", line 1603, in test_ctx_manager
    with psutil.Popen([PYTHON_EXE, "-V"],
  File "c:\cygwin64\home\user\_psutil\psutil\__init__.py", line 1323, in __init__
    self._init(self.__subproc.pid, _ignore_nsp=True)
  File "c:\cygwin64\home\user\_psutil\psutil\__init__.py", line 373, in _init
    self.create_time()
  File "c:\cygwin64\home\user\_psutil\psutil\__init__.py", line 723, in create_time
    self._create_time = self._proc.create_time()
  File "c:\cygwin64\home\user\_psutil\psutil\_pswindows.py", line 678, in wrapper
    return fun(self, *args, **kwargs)
  File "c:\cygwin64\home\user\_psutil\psutil\_pswindows.py", line 924, in create_time
    return cext.proc_create_time(self.pid)
AssertionError: iphr: GetExitCodeProcess() -> STILL_ACTIVE
```

Originates from:
https://github.com/giampaolo/psutil/blob/5140642a7ac95889222cbcc2a8118ca803f0ae3d/psutil/arch/windows/process_utils.c#L103-L108

**Comment by otrejoso** [comment]
I just hit this today in one of my devs machines. The trace is slightly different, but I get the same assertion. Any thoughts?

```
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\project\engine.py:61: in run_project
    run_command(executable, timeout=timeout)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\executor.py:218: in run_command
    LocalExecutor.execute(executable, timeout=timeout)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\executor.py:190: in execute
    NoLogLocalExecutor.execute(executable, timeout)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\performance_functions.py:23: in wrapper
    value = func(*args, **kwargs)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\executor.py:84: in execute
    proc_name, mem, local_children = find_memory_usage(proc)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\testdriver\performance_functions.py:68: in find_memory_usage
    name = proc.name()
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\psutil\__init__.py:731: in name
    name = self._proc.name()
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\psutil\_pswindows.py:707: in wrapper
    return fun(self, *args, **kwargs)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\psutil\_pswindows.py:782: in name
    return py2_strencode(os.path.basename(self.exe()))
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\psutil\_pswindows.py:707: in wrapper
    return fun(self, *args, **kwargs)
C:\depot\bitbucket\project\project\test\tests_pti\.pti_artifactory\venv\lib\site-packages\psutil\_pswindows.py:791: in exe
    exe = cext.proc_exe(self.pid)
E   AssertionError: iphr: GetExitCodeProcess() -> STILL_ACTIVE
```

Not sure if it helps, but I was able to consistently repro it when running things in parallel. A little bit of a background of what we am doing, we run our tests using pytest. We tend to run things in parallel pretty often. We are in the process to drop python 2, and it started to appear when we start to update to python 3.7.

**Comment by giampaolo** [comment]
I should have fixed this in 5.7.0 (still in development).
